srf_generation/source_parameter_generation/rupture_propogation.py

```diff
@@ -0,0 +1,215 @@
+"""
+Rupture Propogation Module
+
+This module provides functions for computing likely rupture paths from
+information about the distances between faults.
+
+Functions:
+    - shaw_dieterich_distance_model:
+      Compute fault jump probabilities using the Shaw-Dieterich distance model.
+    - prune_distance_graph: Prune the distance graph based on a cutoff value.
+    - probability_graph: 
+      Convert a graph of distances between faults into a graph of jump
+      probabilities using the Shaw-Dieterich model.
+    - probabilistic_minimum_spanning_tree: Generate a probabilistic minimum spanning tree.
+    - probabilistic_shortest_path: Generate the shortest path tree from the initial fault.
+
+Typing Aliases:
+    - DistanceGraph: A graph representing distances between faults.
+    - ProbabilityGraph: A graph representing jump probabilities between faults.
+    - RuptureCausalityTree: A tree representing the causality of ruptures between faults.
+"""
+
+from collections import defaultdict
+from typing import Optional
+
+import numpy as np
+
+DistanceGraph = dict[str, dict[str, int]]
+ProbabilityGraph = defaultdict[str, dict[str, float]]
+
+RuptureCausalityTree = dict[str, Optional[str]]
+
+
+def shaw_dieterich_distance_model(distance: float, d0, delta) -> float:
```

Comment:
I assume this Shaw-Dieterich model is from a journal paper? Good to include a reference imo, i.e. title, author, year